reconcile/utils/terrascript_client.py

```diff
@@ -3334,6 +3340,25 @@ def populate_tf_resource_elasticsearch(self, resource, namespace_info):
         for tf_resource in tf_resources:
             self.add_resource(account, tf_resource)
 
+    def _build_es_advanced_security_options(
+            self, advanced_security_options: dict[str, Any]) -> dict[str, Any]:
+        master_user_options = advanced_security_options.pop(
+            'master_user_options', {})
+
+        if master_user_options:
+            master_user_secret = master_user_options['master_user_secret']
+            secret_data = self.secret_reader.read_all(master_user_secret)
+
+            required_keys = {'master_user_name', 'master_user_password'}
+            if secret_data.keys() != required_keys:
```

Comment:
This is even more intuitive than what I had proposed at the beginning.

Reply:
cool!